error while generating release apk

Debug works fine, but when generating release apk I faced this error
```
error: resource android:style/TextAppearance.Material.Widget.Button.Borderless.Colored not found.
error: resource android:style/TextAppearance.Material.Widget.Button.Colored not found.
C:\Users\Ali\.gradle\caches\transforms-1\files-1.1\appcompat-v7-27.1.1.aar\cc1406a2305fbe386bafce46a811510c\res\values-v26\values-v26.xml:9:5-12:13: AAPT: error: resource android:attr/colorError not found.

C:\Users\Ali\.gradle\caches\transforms-1\files-1.1\appcompat-v7-27.1.1.aar\cc1406a2305fbe386bafce46a811510c\res\values-v26\values-v26.xml:13:5-16:13: AAPT: error: resource android:attr/colorError not found.

C:\Users\Ali\.gradle\caches\transforms-1\files-1.1\appcompat-v7-27.1.1.aar\cc1406a2305fbe386bafce46a811510c\res\values-v26\values-v26.xml:17:5-93: AAPT: error: style attribute 'android:attr/keyboardNavigationCluster' not found.

C:\Users\Ali\.gradle\caches\transforms-1\files-1.1\appcompat-v7-27.1.1.aar\cc1406a2305fbe386bafce46a811510c\res\values\values.xml:251:5-69: AAPT: error: resource android:attr/fontStyle not found.

C:\Users\Ali\.gradle\caches\transforms-1\files-1.1\appcompat-v7-27.1.1.aar\cc1406a2305fbe386bafce46a811510c\res\values\values.xml:251:5-69: AAPT: error: resource android:attr/font not found.

C:\Users\Ali\.gradle\caches\transforms-1\files-1.1\appcompat-v7-27.1.1.aar\cc1406a2305fbe386bafce46a811510c\res\values\values.xml:251:5-69: AAPT: error: resource android:attr/fontWeight not found.

error: failed linking references.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-cafe-bazaar:verifyReleaseResources'.
> com.android.ide.common.process.ProcessException: Failed to execute aapt
```
when I updated `app/build.gradle` file, as [this stackoverflow user](https://stackoverflow.com/a/49332191)   said, it did compiled.
```
android {
    compileSdkVersion 26         //was 23
    buildToolsVersion "26.0.1"   //was "23.0.1"
    defaultConfig {
        minSdkVersion 16
        targetSdkVersion 24        //was 22
        versionCode 1
        versionName "1.0"
    }
```
(I updated compileSdkVersion and buildToolsVersion to match my project, maybe it is not needed.)
### I'm using 
`gradle 4.4`
`classpath 'com.android.tools.build:gradle:3.1.2'`
`react-native-cafe-bazaar@1.0.4`
if this can reproduce the problem and the solution works, please update this package.

--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -1,11 +1,15 @@
 apply plugin: "com.android.library"
 
+def safeExtGet(prop, fallback) {
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.1"
+    compileSdkVersion  safeExtGet('compileSdkVersion', 23)
+    buildToolsVersion  safeExtGet('buildToolsVersion', "23.0.1")
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 22
+        targetSdkVersion safeExtGet('targetSdkVersion', 22) 
         versionCode 1
         versionName "1.0"
     }

